### DISCUSSION TITLE
Similar Docs engine?

### DISCUSSION THREAD

**Discussion by milcent** [sepinf-inc/IPED#450] [Q&A] [answered]
What is the engine/algorithm behind the "Find similar documents" functionaloty?
Is it some implementation inside the indexer, like the `MoreLikeThis` from SOLR?
And what is the algorithm used? TF-IDF?
Thanks in advance!

**Comment by lfcnassif** [comment, marked as answer]
Take a look at the updated manual: https://github.com/sepinf-inc/IPED/wiki/User-Manual#similar-document-search

Yes, it uses LUCENE MoreLikeThis internally with some customizations. If you are interested, take a look at the implementation https://github.com/sepinf-inc/IPED/blob/master/iped-engine/src/main/java/dpf/sp/gpinf/indexer/search/SimilarDocumentSearch.java

It is some years old and things could always be improved, help is welcome.

**Reply by lfcnassif** [reply]
There is also this open issue if someone could help: https://github.com/sepinf-inc/IPED/issues/184

**Reply by milcent** [reply]
I commented there. This would be really nice!!
